src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1424,6 +1424,243 @@ public async Task TestRuntimeBaseRouteInNextLinkForPaginatedRestResponse()
             }
         }
 
+        /// <summary>
```

Comment:
Alternative approaches considered for writing tests:

1. Removing the `read` action from `Anonymous` role from any of the existing entities leads to failure of tests. So, alternative options were to a) Have different config file setup for each test while using existing entities (or) b) Introduce new entities without read action in Anonymous role. 
I've gone ahead with having a different config file setup for each test.

2. I tried writing these two tests as a single DataTestMethod as there is an overlap in the mutation and cleanup steps. But that led to lots of conditionals and made the test logic difficult to follow. So, I've written these two as separate tests for better readability.